Gitlab access stopped working

This is probably an issue at my end, but suddenly RepoBee started complaining about

```
[ERROR] either organization git-for-assignments-course2020 could not be found, or the base url 'https://gitlab.tue.nl' is incorrect
```

while running the `setup` or `clone` commands. As I did not update my gitlab settings for the past couple of weeks, this is odd. I checked also with command `repobee -p gitlab config verify`, but it gives similar feedback:

```
GitLabAPI is verifying settings ...
Authenticating connection to https://gitlab.tue.nl...
SUCCESS: Authenticated as XXXXXX at https://gitlab.tue.nl
Trying to fetch group git-for-assignments-course2020
[ERROR] NotFoundError: Could not find group with slug git-for-assignments-course2020. Verify that you have access to the group, and that you've provided the slug (the name in theaddress bar).
[ERROR] A plugin exited with an error
```

I do not understand what is happening: clearly I can still log into gitlab and I do have access to this group via both web and API, it seems. Have you any idea what might be going on?

--- a/src/_repobee/ext/gitlab.py
+++ b/src/_repobee/ext/gitlab.py
@@ -327,19 +327,10 @@ class GitLabAPI(plug.PlatformAPI):
 
     @staticmethod
     def _get_group(group_name, gl):
-        plug.log.debug(f"Searching for groups with name {group_name}")
-        matches = [
-            g
-            for g in gl.groups.list(search=group_name, all=True)
-            if g.path == group_name
-        ]
+        plug.log.debug(f"Fetching group {group_name}")
 
-        if not matches:
-            raise plug.NotFoundError(group_name, status=404)
-
-        plug.log.debug(f"Found groups: {[group.path for group in matches]}")
-
-        return matches[0]
+        with _try_api_request():
+            return gl.groups.get(group_name)
 
     def _get_users(self, usernames):
         users = []

--- a/src/_repobee/ext/gitlab.py
+++ b/src/_repobee/ext/gitlab.py
@@ -94,6 +94,11 @@ class GitLabAPI(plug.PlatformAPI):
 
     def __init__(self, base_url, token, org_name):
         # ssl turns off only for
+
+        plug.log.debug(
+            f"Initializing GitLab API: url={base_url}, target group={org_name}"
+        )
+
         self._user = "oauth2"
         self._gitlab = gitlab.Gitlab(
             base_url, private_token=token, ssl_verify=self._ssl_verify()
@@ -105,7 +110,7 @@ class GitLabAPI(plug.PlatformAPI):
         with _try_api_request():
             self._gitlab.auth()
             self._actual_user = self._gitlab.user.username
-            self._group = self._get_organization(self._group_name)
+            self._group = self._get_group(self._group_name, self._gitlab)
 
     def create_team(
         self,
@@ -320,15 +325,19 @@ class GitLabAPI(plug.PlatformAPI):
             plug.log.warning("SSL verification turned off, only for testing")
         return ssl_verify
 
-    def _get_organization(self, org_name):
+    @staticmethod
+    def _get_group(group_name, gl):
+        plug.log.debug(f"Searching for groups with name {group_name}")
         matches = [
             g
-            for g in self._gitlab.groups.list(search=org_name)
-            if g.path == org_name
+            for g in gl.groups.list(search=group_name, all=True)
+            if g.path == group_name
         ]
 
         if not matches:
-            raise plug.NotFoundError(org_name, status=404)
+            raise plug.NotFoundError(group_name, status=404)
+
+        plug.log.debug(f"Found groups: {[group.path for group in matches]}")
 
         return matches[0]
 
@@ -433,18 +442,7 @@ class GitLabAPI(plug.PlatformAPI):
         user = gl.user.username
 
         plug.echo(f"Trying to fetch group {group_name}")
-        slug_matched = [
-            group
-            for group in gl.groups.list(search=group_name)
-            if group.path == group_name
-        ]
-        if not slug_matched:
-            raise plug.NotFoundError(
-                f"Could not find group with slug {group_name}. Verify that "
-                f"you have access to the group, and that you've provided "
-                f"the slug (the name in the address bar)."
-            )
-        group = slug_matched[0]
+        group = GitLabAPI._get_group(group_name, gl)
         plug.echo(f"SUCCESS: Found group {group.name}")
 
         plug.echo(

--- a/src/_repobee/ext/gitlab.py
+++ b/src/_repobee/ext/gitlab.py
@@ -93,12 +93,6 @@ class GitLabAPI(plug.PlatformAPI):
     _User = collections.namedtuple("_User", ("id", "login"))
 
     def __init__(self, base_url, token, org_name):
-        # ssl turns off only for
-
-        plug.log.debug(
-            f"Initializing GitLab API: url={base_url}, target group={org_name}"
-        )
-
         self._user = "oauth2"
         self._gitlab = gitlab.Gitlab(
             base_url, private_token=token, ssl_verify=self._ssl_verify()

--- a/src/_repobee/ext/gitlab.py
+++ b/src/_repobee/ext/gitlab.py
@@ -45,9 +45,7 @@ def _convert_404_to_not_found_error(msg):
     except gitlab.exceptions.GitlabError as exc:
         if exc.response_code == 404:
             raise plug.NotFoundError(msg)
-        raise plug.UnexpectedException(
-            f"An unexpected exception occured. {type(exc).__name__}: {exc}"
-        )
+        raise plug.PlatformError(str(exc), status=exc.response_code) from exc
 
 
 @contextlib.contextmanager
@@ -63,8 +61,8 @@ def _try_api_request(ignore_statuses: Optional[Iterable[int]] = None):
     """Context manager for trying API requests.
 
     Args:
-        ignore_statuses: One or more status codes to ignore (only
-        applicable if the exception is a gitlab.exceptions.GitlabError).
+        ignore_statuses: One or more status codes to ignore (only applicable if
+            the exception is a gitlab.exceptions.GitlabError).
     """
     try:
         yield
@@ -323,7 +321,9 @@ class GitLabAPI(plug.PlatformAPI):
     def _get_group(group_name, gl):
         plug.log.debug(f"Fetching group {group_name}")
 
-        with _try_api_request():
+        with _convert_404_to_not_found_error(
+            f"could not find group '{group_name}'"
+        ):
             return gl.groups.get(group_name)
 
     def _get_users(self, usernames):

--- a/tests/unit_tests/repobee/plugin_tests/test_gitlab.py
+++ b/tests/unit_tests/repobee/plugin_tests/test_gitlab.py
@@ -510,9 +510,7 @@ class TestVerifySettings:
                 token=TOKEN,
             )
 
-        assert "Could not find group with slug {}".format(
-            non_existing_group
-        ) in str(exc_info.value)
+        assert non_existing_group in str(exc_info.value)
 
     def test_raises_if_master_group_cant_be_found(self):
         non_existing_group = "some-garbage-group"
@@ -525,9 +523,7 @@ class TestVerifySettings:
                 template_org_name=non_existing_group,
             )
 
-        assert "Could not find group with slug {}".format(
-            non_existing_group
-        ) in str(exc_info.value)
+        assert non_existing_group in str(exc_info.value)
 
     def test_raises_when_user_is_not_member(self, mocker):
         gl = GitLabMock(BASE_URL, TOKEN, False)

--- a/tests/unit_tests/repobee/plugin_tests/test_gitlab.py
+++ b/tests/unit_tests/repobee/plugin_tests/test_gitlab.py
@@ -281,11 +281,16 @@ class GitLabMock:
         return list(groups)[: (PAGE_SIZE if not all else None)]
 
     def _get_group(self, id):
-        if id not in self._groups:
-            raise gitlab.exceptions.GitlabGetError(
-                response_code=404, error_message="Group Not Found"
-            )
-        return self._groups[id]
+        if id in self._groups:
+            return self._groups[id]
+
+        for gid, group in self._groups.items():
+            if group.path == id:
+                return group
+
+        raise gitlab.exceptions.GitlabGetError(
+            response_code=404, error_message="Group Not Found"
+        )
 
     def _list_users(self, username=None):
         if username:

